src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -45,71 +47,83 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
+                    if (!(pair.Value is Delegate))
+                    {
+                        var key = pair.Key;
+                        switch(pair.Key.ToLowerInvariant())
                         {
-                            // convert raw strings so they get serialized properly below
-                            content = (string)value;
-                        }
+                            case "body":
+                                // if there is a body, use it as content instead of cleanResponse
+                                content = pair.Value;
+                                break;
 
-                        if (jo.TryGetValue("headers", StringComparison.OrdinalIgnoreCase, out value) && value is JObject)
-                        {
-                            headers = (JObject)value;
-                        }
+                            case "headers":
+                                headers = pair.Value as IDictionary<string, object>;
+                                break;
 
-                        if ((jo.TryGetValue("status", StringComparison.OrdinalIgnoreCase, out value) && value is JValue) ||
-                            (jo.TryGetValue("statusCode", StringComparison.OrdinalIgnoreCase, out value) && value is JValue))
-                        {
-                            statusCode = (HttpStatusCode)(int)value;
+                            case "status":
+                            case "statuscode":
+                                statusCode = (HttpStatusCode)Convert.ToInt32(pair.Value);
+                                break;
                         }
+
+                        // rebuild the responseObject without functions
+                        cleanResponse.Add(pair);
                     }
                 }
-                catch (JsonException)
-                {
-                    // not a json response
-                }
             }
 
+            headers = ConvertKeysToLowercase(headers);
+
             HttpResponseMessage response = CreateResponse(request, statusCode, content, headers);
 
-            if (headers != null)
+            // apply any user specified headers
+            foreach (var header in headers)
             {
-                // apply any user specified headers
-                foreach (var header in headers)
-                {
-                    AddResponseHeader(response, header);
-                }
+                AddResponseHeader(response, header);
             }
 
             request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey] = response;
         }
 
-        private static HttpResponseMessage CreateResponse(HttpRequestMessage request, HttpStatusCode statusCode, object content, JObject headers)
+        private static IDictionary<string, object> ConvertKeysToLowercase(IDictionary<string, object> headers)
+        {
+            IDictionary<string, object> lowercaseHeaders = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Should be able to simplify this and make it a bit more efficient with `headers.ToDictionary`